### PR TITLE
component - z-status-tag - add new component

### PR DESCRIPTION
**Pull request by StefanoMarcolini97**
# component - z-status-tag - add new component

----

### Motivation and Context
Add new graphic element: z-status tag ([story](https://zanichelli.atlassian.net/browse/CLAV-44))

### Priority

- [ ] 1 - Highest
- [ ] 2 - High
- [x] 3 - Medium
- [ ] 4 - Low
- [ ] 5 - Lowest
- [ ] 6 - Not a Priority

### Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] Feature (non-breaking change which adds functionality)
- [x] Component (add a Component as approved dy Design System)
- [ ] Docs (add documentation)
- [ ] Chore (changes that adds small enhancement)
- [ ] Breaking (fix or feature that would cause existing functionality to not work as expected)

### Abstract
[abstract](https://app.abstract.com/share/b8ae04de-5a0f-4ae7-8eab-7829d02a0eb3?mode=design&sha=2bb82d792c779961d592b26c125b5aac7fba35f6)

### Note

----

## Component and Fix approval flow to Master branch

A Pull Request to be merged must have two approvals, one from a member of the product team who developed it and another from a member of the dst dev team other than the team representative.

### CODE REVIEW

File: src/components/z-status-tag/index.tsx [hunk 1]
```diff
@@ -0,0 +1,42 @@
+import { Component, Prop, h, Host } from "@stencil/core";
+import { StatusTagVariant, StatusTagType } from "../../beans";
+import classNames from "classnames";
+
+@Component({
+  tag: "z-status-tag",
+  styleUrl: "styles.css",
+  shadow: false,
```
Comment (DarioSacco) [inline src/components/z-status-tag/index.tsx:8]: why this?

File: src/components/z-status-tag/styles.css [hunk 2]
```diff
@@ -0,0 +1,132 @@
+z-status-tag {
+  --z-icon-width: 14px;
+  --z-icon-height: 14px;
+
+  font-family: var(--dashboard-font);
```
Comment (DarioSacco) [inline src/components/z-status-tag/styles.css:5]: it's better to use the new, clearer variable `--font-family-sans`

File: src/components/z-status-tag/styles.css [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+
+z-status-tag.expandable > span {
+  max-width: 0;
+  transition: max-width 1s;
+  white-space: nowrap;
+  overflow: hidden;
+}
+
+z-status-tag.expandable:hover > span {
+  max-width: 12rem;
```
Comment (DarioSacco) [inline src/components/z-status-tag/styles.css:36]: I haven't found any max width indication in the design sheet 🤔

File: src/components/z-status-tag/styles.css [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+
+z-status-tag.yellow.light {
+  background-color: transparent;
+  color: var(--status-warning-yellow-dark);
+  fill: var(--status-warning-yellow-dark);
+}
+
+z-status-tag.yellow.dark {
+  background-color: var(--status-warning-yellow-dark);
+  color: var(--text-white);
```
Comment (DarioSacco) [inline src/components/z-status-tag/styles.css:54]: text color in the dark variants is not white (except for the neutral/gray variant). for example the text over the warning-dark variant is `#fbf4e6` (`--yellow50` in the new variables, `--status-warning-yellow-lighter` in the legacy ones)

**Review by DarioSacco — CHANGES_REQUESTED** [review]

File: src/components/z-status-tag/styles.css [hunk 5]
```diff
@@ -0,0 +1,132 @@
+z-status-tag {
+  --z-icon-width: 14px;
+  --z-icon-height: 14px;
+
+  font-family: var(--dashboard-font);
+  font-weight: var(--font-rg);
```
Comment (DarioSacco) [inline src/components/z-status-tag/styles.css:6]: this can be removed

**Review by DarioSacco — CHANGES_REQUESTED** [review]

File: src/components/z-status-tag/styles.css (hunk 3 above)
Comment (adecastri) [reply src/components/z-status-tag/styles.css:36]: Max-width is used for the animation, to show/hide text

Comment (DarioSacco) [reply src/components/z-status-tag/styles.css:36]: you can set `100%`

File: src/components/z-status-tag/styles.css [hunk 6]
```diff
@@ -0,0 +1,132 @@
...
+}
+
+z-status-tag:not(.expandable) > z-icon:not(:last-child),
+z-status-tag.expandable:hover > z-icon:not(:last-child) {
+  margin-right: var(--space-unit);
+}
+
+z-status-tag.expandable > span {
+  max-width: 0;
+  transition: max-width 1s;
```
Comment (DarioSacco) [inline src/components/z-status-tag/styles.css:30]: the transition duration is different from the abstract design document

**Review by DarioSacco — CHANGES_REQUESTED** [review]

File: src/components/z-status-tag/index.tsx [hunk 7]
```diff
@@ -30,9 +30,10 @@ export class ZStatusTag {
   render() {
     return (
       <Host
-        class={classNames(this.type, this.variant, {
+        class={classNames(this.variant, {
           expandable: this.expandable && this.icon && this.text,
         })}
+        status={this.status}
```
Comment (DarioSacco) [inline src/components/z-status-tag/index.tsx:36]: this is not necessary, `{ reflect: true }` does this for you

**Review by DarioSacco — CHANGES_REQUESTED** [review]

File: src/index.html [hunk 8]
```diff
@@ -988,6 +988,11 @@ <h3>Item 3</h3>
 
   <br /><br /><br /><br /><br /><br />
 
+  <div style="padding-left: 280px;">
+    <z-status-tag icon="gear" text="z-status-tag" type="warning" variant="dark" expandable="true"></z-status-tag>
```
Comment (DarioSacco) [inline src/index.html:992]: `type` is now `status`

**Review by DarioSacco — CHANGES_REQUESTED** [review]

File: src/beans/index.tsx [hunk 9]
```diff
@@ -235,3 +235,17 @@ export enum SortingOrientation {
   desc = "desc",
   none = "none",
 }
+
+export enum StatusTagVariant {
```
[the remark this answers is not in the document]
Comment (adecastri) [reply src/beans/index.tsx]: risolto

File: src/components/z-status-tag/index.tsx (hunk 1 above)
Comment (adecastri) [reply src/components/z-status-tag/index.tsx:8]: Lasciando **shadow: true** sembra non si possa utilizzare il tag che si scrive e si deve per forza aggiungere un elemento che contenga testo e icona con le proprietà adesso assegnate allo z-status-tag

File: src/components/z-status-tag/styles.css (hunk 2 above)
Comment (adecastri) [reply src/components/z-status-tag/styles.css]: risolto

File: src/components/z-status-tag/styles.css (hunk 4 above)
Comment (adecastri) [reply src/components/z-status-tag/styles.css]: risolto

File: src/components/z-status-tag/styles.css (hunk 5 above)
Comment (adecastri) [reply src/components/z-status-tag/styles.css]: risolto

File: src/components/z-status-tag/index.tsx (hunk 7 above)
Comment (adecastri) [reply src/components/z-status-tag/index.tsx:36]: Usando il tag Host nel render il reflect non funziona se si scrive direttamente il tag del web component. Facciamo altre prove, casomai togliamo reflect

File: src/components/z-status-tag/styles.css (hunk 3 above)
Comment (adecastri) [reply src/components/z-status-tag/styles.css]: Abbiamo provato con 100% ma si spacca l'animazione, succede una cosa brutta: il testo si riduce ma continua ad accupare lo spazio fino a fine animazione

File: src/components/z-status-tag/styles.css (hunk 6 above)
Comment (adecastri) [reply src/components/z-status-tag/styles.css]: risolto

File: src/components/z-status-tag/index.tsx (hunk 7 above)
Comment (adecastri) [reply src/components/z-status-tag/index.tsx]: risolto

File: src/index.html (hunk 8 above)
Comment (adecastri) [reply src/index.html]: risolto

File: src/components/z-status-tag/styles.css (hunk 3 above)
Comment (DarioSacco) [reply src/components/z-status-tag/styles.css]: forse con `max-width: unset`? in ogni caso settare una max-width fissa non rispetta le specifiche del design

**Review by DarioSacco — APPROVED** [review]

**Review by francescochiapella — APPROVED** [review]

**Pull request merged**